### PR TITLE
yasm: fix test for Linux

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/yasm.rb [hunk 1]
```diff
@@ -39,24 +39,57 @@ def install
...
+      int 0x80
     EOS
-    system "#{bin}/yasm", "-f", "macho64", "test.asm"
-    system "/usr/bin/ld", "-macosx_version_min", "10.8.0", "-static", "-o", "test", "test.o"
-    system "./test"
+    system "#{bin}/yasm", "foo.s"
+    code = File.open("foo", "rb") { |f| f.read.unpack("C*") }
+    expected = [0x66, 0xb8, 0x00, 0x00, 0x00, 0x00, 0x66, 0xbb,
+                0x00, 0x00, 0x00, 0x00, 0xcd, 0x80]
+    assert_equal expected, code
```
Comment (cho-m) [inline Formula/yasm.rb:51]: Extra test copied from [`nasm`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/nasm.rb).

File: Formula/yasm.rb [hunk 2]
```diff
@@ -39,24 +39,57 @@ def install
...
+            mov     rdi, 0
+            syscall
+        section .data
+        msg:    db      "Hello, world!", 10
+        .len:   equ     $ - msg
+      EOS
+      system "#{bin}/yasm", "-f", "elf64", "test.asm"
+      system "/usr/bin/ld", "-static", "-o", "test", "test.o"
+    end
+    assert_equal "Hello, world!\n", shell_output("./test")
```
Comment (cho-m) [inline Formula/yasm.rb:93]: Tried to add ELF64-equivalent assembly for Linux and updated check to test for equality of output test.

**Review by danielnachun — COMMENTED** [review]
Really nice work adding better tests to this.  Would it make sense for our assembler formulae to add ARM64 tests on macOS?  I'm not sure if that's been looked into yet.

**Comment by cho-m** [comment]
I don't think `yasm` supports ARM64 as an architecture.
```
❯ yasm -a help
Available yasm architectures:
    x86         x86 (IA-32 and derivatives), AMD64
    lc3b        LC-3b
```
Probably similar situation for `nasm` as they are similar.

**Review by danielnachun — APPROVED** [review]
I seen now that `nasm`/`yasm` is explicitly x86-only.  In that case there's nothing more to add here.

**Pull request merged**